ios/FluentUI/Navigation/NavigationBarTokenSet.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// Design token set for the `NavigationBar` control.
+public class NavigationBarTokenSet: ControlTokenSet<NavigationBarTokenSet.Tokens> {
+    public enum Tokens: TokenSetKey {
+        /// Describes the background color for the navigation bar.
+        case backgroundColor
+
+        /// Describes the color of the buttons at the top of the navigation bar.
+        case buttonTintColor
+
+        /// Describes the font used for a "large" title.
+        case largeTitleFont
+
+        /// Describes the color of the subtitle.
+        case subtitleColor
+
+        /// Describes the font used for the subtitle.
+        case subtitleFont
+
+        /// Describes the color of the title.
+        case titleColor
+
+        /// Describes the font used for a "small" title.
+        case titleFont
+    }
+
+    init(style: @escaping () -> NavigationBar.Style) {
+        super.init { [style] token, theme in
+            switch token {
+            case .backgroundColor:
+                return .uiColor {
+                    switch style() {
+                    case .primary, .default, .custom:
+                        return UIColor(light: theme.color(.brandBackground1).light,
+                                       dark: theme.color(.background3).dark)
+                    case .system:
+                        return theme.color(.background3)
+                    }
+                }
+            case .buttonTintColor, .subtitleColor: // By default, these return the same color
+                return .uiColor {
+                    switch style() {
+                    case .primary, .default, .custom:
+                        return UIColor(light: theme.color(.foregroundOnColor).light,
+                                       dark: theme.color(.foreground2).dark)
+                    case .system:
+                        return theme.color(.foreground2)
+                    }
+                }
+            case .largeTitleFont:
+                return .uiFont {
+                    theme.typography(.title1)
+                }
+            case .subtitleFont:
+                return .uiFont {
+                    theme.typography(.caption1)
+                }
+            case .titleColor:
```

Comment:
Is the issue of the titleColor token not being reflected on the nav bar fixed?

Reply:
I'm not sure what you mean by this, but I don't see any issues getting the titleColor token to show up. If there's still a bug, please provide a set of repro steps.

Reply:
For example, if you change the titleColor token to red, does the title in the nav bar show up as red?

Reply:
Aha, I see. Even if we change the default color to red, the visible title `Label` is still black because it's being driven by the `Label`'s `TextColorStyle` of `.regular`.

The proper fix for this would be to get rid of `TextColorStyle` altogether (we've already gotten the OK to do this) so we can just pass in the default token directly. However, token overrides still work fine, and I'm not sure if such a change should be baked into this PR, so we're better off doing this separately.

The question now is, do we consider this a blocking bug?

<img width="761" alt="Screenshot 2023-05-05 at 4 54 56 PM" src="https://user-images.githubusercontent.com/717674/236586292-77a37b13-8db5-4aad-b39f-a4603a4a638c.png">

Reply:
Should probably be fixed before we merge this PR and cut 17.

Reply:
This will be fixed with https://github.com/amgleitman/fluentui-apple/pull/2.